README.md

```diff
@@ -1,11 +1,124 @@
 # Affected-Paths
 
-Affected-Paths is a Gradle specific tool that allows parsing Gradle builds and identifies all projects affected
-(directly and indirectly) given a list of file changes between two git commits.
+Affected-Paths is a Java library that utilizes the Gradle Tooling API to parse Gradle based projects and identifies all modules affected
+(directly and indirectly) given the file changes from git.
 
-## Getting Started
+## Quick Demo
 
-In order to use the `affected-paths`, the `tooling-support` Gradle plugin must be applied to every project `build.gradle`.
+To get started quickly with the demo app, first build the distribution of it:
+```shell
+./gradlew :affected-paths:app:installDist
+```
+
+Once the distribution has been built, the demo app can be run on a given project by calling:
+```shell
+./affected-paths/app/build/install/affected-paths/bin/affected-paths --log-gradle --inject-plugin --dir=/path/to/project
```

Comment:
I get 
```
pushd ~/Development
git clone https://github.com/square/workflow-kotlin.git
popd
./affected-paths/app/build/install/affected-paths/bin/affected-paths --log-gradle --inject-plugin --dir=$HOME/Development/workflow-kotlin/samples/tictactoe/app  
```

I see:

```
Caused by: java.lang.IllegalArgumentException: No known plugin used (should be Android or Java)
	at com.squareup.tooling.support.builder.SquareProjectModelBuilderImpl.buildAll(SquareProjectModelBuilder.kt:48)
```

Reply:
Also adding the `--debug-gradle` is helpful. 

<img width="1035" alt="image" src="https://github.com/square/affected-paths/assets/326857/1b172cf4-62a2-4b9f-8ab7-842313b97d00">

For some reason the only project in that repository that applies this plugin is `:benchmarks`:

<img width="891" alt="image" src="https://github.com/square/affected-paths/assets/326857/fbc2c7af-051c-42b0-9f00-c899e2f547b1">

Reply:
> I tried:
> 
> ```
> pushd ~/Development
> git clone https://github.com/square/workflow-kotlin.git
> popd
> ./affected-paths/app/build/install/affected-paths/bin/affected-paths --log-gradle --inject-plugin --dir=$HOME/Development/workflow-kotlin/samples/tictactoe/app  
> ```
> 
> I see:
> 
> ```
> Caused by: java.lang.IllegalArgumentException: No known plugin used (should be Android or Java)
> 	at com.squareup.tooling.support.builder.SquareProjectModelBuilderImpl.buildAll(SquareProjectModelBuilder.kt:48)
> ```

This is a bug I also found. I fixed it with this [commit](https://github.com/square/affected-paths/pull/11/commits/73efc29c44b4d2c3918938535791f5ca54c28e9e). The fix won't be available until the release, but to test it locally, you can do the following:

- In the `gradle.properties`, set the `RELEASE_SIGNING_ENABLED` flag to `false`
- Run `./gradlew -p tooling/support publishToMavenLocal`
- In the init script, add in `mavenLocal()` to the `buildScript.repositories` lambda.

After following the above, the steps in the readme should work.

Reply:
I also needed `./gradlew :affected-paths:app:installDist` after making the change in the init script.

Works great!

```
Changed file: workflow-testing/api/workflow-testing.api
Projects affected by this changed file:
    - :samples:dungeon:timemachine:debug:debugUnitTest
    - :samples:dungeon:timemachine:release:releaseUnitTest
    - :samples:tictactoe:common:debug:debugUnitTest
    - :samples:tictactoe:common:release:releaseUnitTest
    - :workflow-rx2:debug:debugUnitTest
    - :workflow-rx2:release:releaseUnitTest
    - :workflow-testing

Changed file: workflow-testing/build.gradle.kts
Projects affected by this changed file:
    - :samples:dungeon:timemachine:debug:debugUnitTest
    - :samples:dungeon:timemachine:release:releaseUnitTest
    - :samples:tictactoe:common:debug:debugUnitTest
    - :samples:tictactoe:common:release:releaseUnitTest
    - :workflow-rx2:debug:debugUnitTest
    - :workflow-rx2:release:releaseUnitTest
    - :workflow-testing

Changed file: workflow-testing/dependencies/runtimeClasspath.txt
Projects affected by this changed file:
    - :samples:dungeon:timemachine:debug:debugUnitTest
    - :samples:dungeon:timemachine:release:releaseUnitTest
    - :samples:tictactoe:common:debug:debugUnitTest
    - :samples:tictactoe:common:release:releaseUnitTest
    - :workflow-rx2:debug:debugUnitTest
    - :workflow-rx2:release:releaseUnitTest
    - :workflow-testing

Changed file: workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkerTester.kt
Projects affected by this changed file:
    - :samples:dungeon:timemachine:debug:debugUnitTest
    - :samples:dungeon:timemachine:release:releaseUnitTest
    - :samples:tictactoe:common:debug:debugUnitTest
    - :samples:tictactoe:common:release:releaseUnitTest
    - :workflow-rx2:debug:debugUnitTest
    - :workflow-rx2:release:releaseUnitTest
    - :workflow-testing
    - :workflow-testing:debug:debugUnitTest
    - :workflow-testing:release:releaseUnitTest

Changed file: workflow-testing/src/test/java/com/squareup/workflow1/WorkerTest.kt
Projects affected by this changed file:
    - :workflow-testing:debug:debugUnitTest
    - :workflow-testing:release:releaseUnitTest

Changed file: workflow-testing/src/test/java/com/squareup/workflow1/testing/WorkerTesterTest.kt
Projects affected by this changed file:
    - :workflow-testing:debug:debugUnitTest
    - :workflow-testing:release:releaseUnitTest

17:54:00.011 [main] INFO  c.squareup.affected.paths.app.Main -- Process ran for 17.856227459s
```